### PR TITLE
Separate functional tests

### PR DESCRIPTION
**Pull request by mateuszkwiecinski**
In reference to this comment https://github.com/jeremymailen/kotlinter-gradle/pull/124#discussion_r368681815 I've started playing around with the code.

I'm creating PR to get some feedback if that's the separation you had in mind, or the functional tests should be separated even furhter.

Currently the plugin is added to TestKit projects buildscript using existing mechanism involving pluginUnderTestMetadata task, but each project has its own setup. 
I wasn't sure if it is possible to achieve more separated environment using TestKit library 🤔When I've been trying to add completely isolated plugin, but still keep it as a part of the build (for the ease of maintenance and further development) I faced issue where resolved dependency classes were loaded using different classloaders.

FYI: I'll rewrite all scripts to `kts` after #127 is merged.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -31,11 +31,6 @@ dependencies {
     testImplementation 'org.jetbrains:annotations:18.0.0'
 }
 
-// Required to put all `compileOnly` dependencies on the classpath for the functional test suite
-tasks.withType(PluginUnderTestMetadata).configureEach {
-    pluginClasspath.from(configurations.compileOnly)
-}
-
```
Comment (mateuszkwiecinski) [inline build.gradle]: My assumption was that the goal of this PR is to remove this few lines.

File: build.gradle [hunk 2]
```diff
@@ -78,3 +73,5 @@ publishing {
 wrapper {
     gradleVersion = '6.0.1'
 }
+tasks.register("noOp")
+defaultTasks "noOp"
```
Comment (mateuszkwiecinski) [inline build.gradle]: Hope that doesn't bother anyone. By default gradle invokes `help` task and that unnecessarily bloats console output

File: functional/android/build.gradle [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+repositories {
+    jcenter()
+    google()
+}
+
+dependencies {
+    runtimeOnly project(':')
+    runtimeOnly 'com.android.tools.build:gradle:3.5.3'
+    runtimeOnly 'org.jetbrains.kotlin:kotlin-gradle-plugin'
```
Comment (mateuszkwiecinski) [inline functional/android/build.gradle]: I could be using `testRuntimeOnly` but adding it to `runtimeOnly` make it gets picked by `java-gradle-plugin`'s `pluginUnderTestMetadata` task and automatically applied to TestKit invoked projects.

File: functional/build.gradle [hunk 4]
```diff
@@ -0,0 +1,16 @@
+plugins {
+    id 'org.jetbrains.kotlin.jvm'
+    id 'org.jmailen.kotlinter'
+    id 'java-test-fixtures'
```
Comment (mateuszkwiecinski) [inline functional/build.gradle]: I decided to give a try to the test fixtures plugin to emphasize that all the sources here are meant to be used in tests sources only.

File: functional/kotlin/build.gradle [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+repositories {
+    jcenter()
+    google()
+}
+
+dependencies {
+    runtimeOnly project(':')
+    runtimeOnly 'org.jetbrains.kotlin:kotlin-gradle-plugin'
```
Comment (mateuszkwiecinski) [inline functional/kotlin/build.gradle]: I verified that this setup would catch the mistake I did in https://github.com/jeremymailen/kotlinter-gradle/pull/124#discussion_r368314350

**Comment by jeremymailen** [comment]
Thank you! I'll review later in the week, but it seems like a promising approach.

**Pull request closed without merging**